tests/unit/parallel/lexicographical_compare.cpp

```diff
@@ -0,0 +1,475 @@
+
+    test_lexicographical_compare3(execution_policy(seq), IteratorTag());
+    test_lexicographical_compare3(execution_policy(par), IteratorTag());
+    test_lexicographical_compare3(execution_policy(par_vec), IteratorTag());
+    test_lexicographical_compare3(execution_policy(seq(task)), IteratorTag());
+    test_lexicographical_compare3(execution_policy(par(task)), IteratorTag());
+}
+
+void lexicographical_compare_test3()
+{
+    test_lexicographical_compare3<std::random_access_iterator_tag>();
+    test_lexicographical_compare3<std::forward_iterator_tag>();
+    test_lexicographical_compare3<std::input_iterator_tag>();
+}
+
+///////////////////////////////////////////////////////////////////////////////
+template <typename ExPolicy, typename IteratorTag>
+void test_lexicographical_compare_exception(ExPolicy const& policy, IteratorTag)
+{
+    BOOST_STATIC_ASSERT(hpx::parallel::is_execution_policy<ExPolicy>::value);
+
+    typedef std::vector<std::size_t>::iterator base_iterator;
+    typedef test::decorated_iterator<base_iterator, IteratorTag>
+        decorated_iterator;
+
+    std::vector<std::size_t> c(10007);
+    std::iota(boost::begin(c), boost::end(c), 0);
+
+    std::vector<std::size_t> h(10007);
+    std::iota(boost::begin(h), boost::end(h), 0);
+
+    bool caught_exception = false;
+    try {
+        hpx::parallel::lexicographical_compare(policy,
+            decorated_iterator(
+                boost::begin(c),
+                [](){ throw std::runtime_error("test"); }),
+            decorated_iterator(
+                boost::end(c),
+                [](){ throw std::runtime_error("test"); }),
+            boost::begin(h), boost::end(h));
+        HPX_TEST(false);
+    }
+    catch(hpx::exception_list const& e) {
+        caught_exception = true;
+        test::test_num_exceptions<ExPolicy, IteratorTag>::call(policy, e);
+    }
+    catch(...) {
+        HPX_TEST(false);
+    }
+
+    HPX_TEST(caught_exception);
+}
+
+template <typename ExPolicy, typename IteratorTag>
+void test_lexicographical_compare_async_exception(ExPolicy const& p, IteratorTag)
+{
+    typedef std::vector<std::size_t>::iterator base_iterator;
+    typedef test::decorated_iterator<base_iterator, IteratorTag>
+        decorated_iterator;
+
+    std::vector<std::size_t> c(10007);
+    std::fill(boost::begin(c), boost::end(c), std::rand() + 1);
+
+    std::vector<std::size_t> h(10006);
+    std::fill(boost::begin(h), boost::end(h), std::rand() + 1);
+
+    bool caught_exception = false;
+    try {
+        hpx::future<bool> f =
+            hpx::parallel::lexicographical_compare(p,
+                decorated_iterator(
+                    boost::begin(c),
+                    [](){ throw std::runtime_error("test"); }),
+                decorated_iterator(
+                    boost::end(c),
+                    [](){ throw std::runtime_error("test"); }),
+            boost::begin(h), boost::end(h));
+        f.get();
```

Comment:
add a test that the algorithm actually returned before calling `f.get()` (see here for an example: https://github.com/STEllAR-GROUP/hpx/blob/master/tests/unit/parallel/copy.cpp#L202)